### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by cooperlees**
- Add MacOS testing
- Adds 3.7 testing for Windows

Azure test run: https://dev.azure.com/cooper0394/cooper/_build/results?buildId=6
- Windows tests fail as expected as reported on #728 ...

### CODE REVIEW

**Comment by illfated** [comment]
Maybe cosmetic, but it looks like a NewLine needs to be added at the end of the file.

**Comment by cooperlees** [comment]
Yeah, not to sure why i didn’t get any Travis action.

In regards to the new line, the yaml worked on my fork. I think @ambv just needs to click some GitHub UI to give Azure access and it should work ...

**Comment by illfated** [comment]
Could also be that my comment was redundant, if black doesn't have any codestyle guidelines saying that all files should end with a newline. I guess I am so used to scenarios where source files need to be added sequentially to build some form of DB content, that it could potentially be code breaking to skip NewLine at the end of each file. For all I know, it does not matter at all for this repository.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+trigger:
+- master
```
Comment (hugovk) [inline azure-pipelines.yml]: Maybe remove this so feature branches are tested too?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+
+- job: 'Testing_black_Mac'
```
Comment (hugovk) [inline azure-pipelines.yml]: Black is a proper noun, just like Mac and Windows.
```suggestion
- job: 'Testing_Black_Mac'
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+        ci.env: 'py36'
+      Python37:
+        python.version: '3.7'
+        ci.env: 'py37'
+
+- job: 'Testing_black_Windows'
```
Comment (hugovk) [inline azure-pipelines.yml]:
```suggestion
- job: 'Testing_Black_Windows'
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        ci.env: 'py37'
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '$(python.version)'
+      architecture: 'x64'
+
+  - script: python setup.py test
+    displayName: 'Running black tests for $(ci.env)'
```
Comment (hugovk) [inline azure-pipelines.yml]:
```suggestion
    displayName: 'Running Black tests for $(ci.env)'

```

File: azure-pipelines.yml (hunk 1 above)
Comment (cooperlees) [reply azure-pipelines.yml]: Sure can. I rely on travis for that noise on my projects.

**Comment by cooperlees** [comment]
Rightio - Will get your very important capitalization fixes up.

**Comment by cooperlees** [comment]
I have no idea how adding a .yml file brought the coverage down - This seems flaky!

Azure CI failed as I expected: https://github.com/cooperlees/black/runs/75546440

**Comment by ambv** [comment]
Thanks Cooper, we went with #1085 instead.

**Comment by cooperlees** [comment]
A worthy alternative. No problems.

**Pull request closed without merging**